### PR TITLE
Retire the gpkerberos test

### PR DESCRIPTION
**Pull request by adam8157**
Upstream has the 'kerberos' tap test, this commit adds some tests, which
only Greenplum has, into it, and retire the gpkerberos.

### CODE REVIEW

File: src/test/kerberos/t/001_auth.pl [hunk 1]
```diff
@@ -276,6 +276,47 @@ sub test_access
 	"gssencmode=disable",
 	"succeeds with GSS encryption disabled and hostnogssenc hba");
 
+unlink($node->data_dir . '/pg_hba.conf');
```
Comment (asimrp) [inline src/test/kerberos/t/001_auth.pl:279]: Can we please add a comment explaining why `pg_hba.conf` needs to be rewritten?

File: src/test/kerberos/t/001_auth.pl [hunk 2]
```diff
@@ -276,6 +276,47 @@ sub test_access
...
+$node->append_conf('pg_hba.conf',
+	qq{host all all $hostaddr/32 trust});
+$node->restart;
+
+# Greenplum tests for expiration, remove if upstream adds the similar tests
```
Comment (asimrp) [inline src/test/kerberos/t/001_auth.pl:284]: Does it make sense to submit a patch?

**Review by asimrp — APPROVED** [review]
+1 to remove redundant test and leverage existing TAP tests.

File: src/test/kerberos/t/001_auth.pl (hunk 1 above)
Comment (adam8157) [reply src/test/kerberos/t/001_auth.pl:281]: Done, thanks.

File: src/test/kerberos/t/001_auth.pl (hunk 2 above)
Comment (adam8157) [reply src/test/kerberos/t/001_auth.pl]: Sure, I will give it a try.

**Pull request merged**